che-incubator/che-code
# fix: make workspace restart more informative

### What does this PR do?
- bumps to `@eclipse-che/che-devworkspace-generator:7.86.0`
- displays a reason of a failure when trying to restart the workspace from a local devfile
- allows to easily open the devfile by a notification button

![Screenshot from 2024-05-28 18-55-59](https://github.com/che-incubator/che-code/assets/1655894/0adcab92-1b56-464b-9ba1-1607089e5380)

### What issues does this PR fix?

https://github.com/eclipse-che/che/issues/22913

### How to test this PR?
- create a repository with https://github.com/vitaliy-guliy/vscode-test-extension?editor-image=quay.io/che-incubator-pull-requests/che-code:pr-363-amd64
- install yaml extension
- open `.devfile.yaml` and make some invalid change. I usually add the container environment variable
```
      env:
        - name: test
          value: true
```
The `yaml` extension should underline `true` as one that has wrong type (considering devfile schema it should be string instead of boolean).

- run `Eclipse Che: Restart Workspace from Local Devfile` command and select the devfile
( if there is only one devfile in the project, you can press `Enter` several times to confirm the reload )

- at this step you should be warned with an explanation of the reason of failure

### Does this PR contain changes that override default upstream Code-OSS behavior?
